chat-android/src/main/java/com/ably/chat/Typing.kt

```diff
@@ -178,18 +178,19 @@ internal class DefaultTyping(
         logger.trace("DefaultTyping.stop()")
         typingScope.launch {
             typingJob?.cancel()
-            channel.presence.leaveClientCoroutine(clientId)
+            room.ensureAttached()
+            channel.presence.leaveClientCoroutine(room.clientId)
         }.join()
```

Comment:
No this needs to be a part of `typingScope` which is `CoroutineScope` with `parallalism` set to 1 to avoid race conditions while setting and cancelling `typingJob`